projects/components/src/action-menu/action-search.provider.spec.ts

```diff
@@ -0,0 +1,99 @@
+}
+
+const MOCK_DATA: ActionItem<any, any>[] = [
+    {
+        textKey: 'Contextual 1',
+        handler: () => console.log('Contextual 1'),
+    },
+    {
+        textKey: 'power.actions',
+        children: [
+            {
+                textKey: 'Start',
+                handler: () => console.log('Starting '),
+            },
+            {
+                textKey: 'Stop',
+                handler: () => console.log('Stopping '),
+            },
+        ],
+    },
+];
+
+describe('ActionSearchProvider', () => {
+    beforeEach(function(this: HasActionSearchProvider): void {
+        this.actionSearchProvider = new ActionSearchProvider(
+            new SpotlightSearchService(),
+            new MockTranslationService()
+        );
+    });
+    describe('actions', () => {
+        it('updates the flattened list of available actions when set', function(this: HasActionSearchProvider): void {
+            expect((this.actionSearchProvider as any).flatListOfAvailableActions).toEqual([]);
+            this.actionSearchProvider.actions = MOCK_DATA;
+            expect((this.actionSearchProvider as any).flatListOfAvailableActions).toEqual(
+                (this.actionSearchProvider as any).getFlatListOfAvailableActions(MOCK_DATA)
+            );
+        });
+    });
+
+    describe('selectedEntities', () => {
+        it('calls the updateFlatListOfAvailableActions method when set', function(this: HasActionSearchProvider): void {
+            const spy = spyOn(this.actionSearchProvider as any, 'updateFlatListOfAvailableActions').and.callFake(
+                () => null
+            );
+            this.actionSearchProvider.selectedEntities = [];
+            expect(spy).toHaveBeenCalled();
+        });
+    });
+
+    describe('actionProviderName', () => {
+        it('calls the register method when set', function(this: HasActionSearchProvider): void {
+            const spy = spyOn(this.actionSearchProvider, 'register').and.callFake(() => null);
+            this.actionSearchProvider.actionProviderName = 'blah';
+            expect(spy).toHaveBeenCalled();
+        });
+    });
+
+    describe('register', () => {
+        it('calls the registerProvider method of spotlight search service', function(this: HasActionSearchProvider): void {
+            const spy = spyOn((this.actionSearchProvider as any).spotlightSearchService, 'registerProvider');
+            this.actionSearchProvider.register();
+            expect(spy).toHaveBeenCalledWith(
+                this.actionSearchProvider,
+                '{"key":"vcd.cc.action.provider.section.title","params":[{"actionProviderName":""}]}'
+            );
+        });
+
+        it(
+            'calls the registerProvider method of spotlight search service if registration id exists' + ' already',
+            function(this: HasActionSearchProvider): void {
+                const spy = spyOn((this.actionSearchProvider as any).spotlightSearchService, 'unregisterProvider');
```

Comment:
Do not get the spotlightSearchService from the private property of the provider. In general getting access to private properties or methods should be a very rare case